stdpopsim/slim_engine.py

```diff
@@ -848,9 +848,16 @@ def matrix2str(
 
     # Mutation types.
     for i, m in enumerate(mutation_types):
-        distrib_args = [str(arg) for arg in m.distribution_args]
-        distrib_args[m.Q_scaled_index] = "Q * " + distrib_args[m.Q_scaled_index]
-        distrib_args = ", ".join(distrib_args)
+        if len(m.Q_scaled_index) >= 1:
+            distrib_args = [str(arg) for arg in m.distribution_args]
+            for i in m.Q_scaled_index:
```

Comment:
I reckon the tests are failing because this loop variable `i` is stomping on the outer loop variable which is also named `i`.